application/di/init.go

```diff
@@ -93,27 +97,39 @@ func initDomain() {
 }
 
 func initInfrastructure() {
-	config.CurrentConfig().AddBinaryLocationsToPath(
-		[]string{
-			filepath.Join(xdg.Home, ".sdkman"),
-			"/usr/lib",
-			"/usr/java",
-			"/opt",
-			"/Library",
+	c := config.CurrentConfig()
+	//goland:noinspection GoBoolExpressions
+	if runtime.GOOS == "windows" {
+		// on windows add the locations in the background, as it can take a while and shouldn't block the server
+		go c.AddBinaryLocationsToPath([]string{
 			"C:\\Program Files",
 			"C:\\Program Files (x86)",
-		},
-	)
+		})
+	} else {
+		c.AddBinaryLocationsToPath(
+			[]string{
+				filepath.Join(xdg.Home, ".sdkman"),
+				"/usr/lib",
+				"/usr/java",
+				"/opt",
+				"/Library",
+			})
+	}
 
+	initializeWorkflowEngine()
 	errorReporter = sentry2.NewSentryErrorReporter()
-	installer = install.NewInstaller(errorReporter)
+	installer = install.NewInstaller(errorReporter, engine.GetNetworkAccess().GetUnauthorizedHttpClient)
```

Comment:
Just FYI, I think it would it would have also been fine to use the GetHttpClient() here.